src/main/kotlin/no/nav/personbruker/dittnav/eventtestproducer/ytelsestesting/TestDataService.kt

```diff
@@ -0,0 +1,92 @@
+package no.nav.personbruker.dittnav.eventtestproducer.ytelsestesting
+
+import no.nav.personbruker.dittnav.eventtestproducer.beskjed.BeskjedProducer
+import no.nav.personbruker.dittnav.eventtestproducer.beskjed.ProduceBeskjedDto
+import no.nav.personbruker.dittnav.eventtestproducer.common.InnloggetBruker
+import no.nav.personbruker.dittnav.eventtestproducer.common.createKeyForEvent
+import no.nav.personbruker.dittnav.eventtestproducer.done.DoneProducer
+import no.nav.personbruker.dittnav.eventtestproducer.innboks.InnboksProducer
+import no.nav.personbruker.dittnav.eventtestproducer.innboks.ProduceInnboksDto
+import no.nav.personbruker.dittnav.eventtestproducer.oppgave.OppgaveProducer
+import no.nav.personbruker.dittnav.eventtestproducer.oppgave.ProduceOppgaveDto
+import org.slf4j.LoggerFactory
+import java.time.Instant
+
+class TestDataService(
+        private val doneProducer: DoneProducer,
+        private val beskjedProducer: BeskjedProducer,
+        private val oppgaveProducer: OppgaveProducer,
+        private val innboksProducer: InnboksProducer
+) {
+
+    private val log = LoggerFactory.getLogger(TestDataService::class.java)
+
+    private val bruker = InnloggetBruker("88888", 4, "dummyToken")
+    private val dummySystembruker = "dittnav"
+
+    private val antallEventer = 50000
+
+    fun produserBeskjederOgTilhorendeDoneEventer() {
+        log.info("Produserer $antallEventer beskjeder")
+        val start = Instant.now()
+        for (i in 1..antallEventer) {
+            val key = createKeyForEvent("b-$i", dummySystembruker)
+            val dto = ProduceBeskjedDto("Beskjedtekst $i", "https://beskjed-$i")
+            val beskjedEvent = beskjedProducer.createBeskjedForIdent(bruker, dto)
+            val doneEvent = doneProducer.createDoneEvent(bruker)
+            doneProducer.produceEvent(bruker, key, doneEvent)
+            beskjedProducer.produceEvent(bruker, key, beskjedEvent)
+            if (isShouldTakeASmallBreakAndLogProgress(i)) {
+                log.info("Har produsert beskjed nummer $i tar en liten pause")
+                Thread.sleep(1000)
```

Comment:
Ikke spesielt viktig mtp. det er en test-app, men hvis du slenger på 'suspend' på funksjonen kan du bruke delay() i stedet for

Reply:
Ja, enig, jeg bytter til `delay()`.